greengrocer_class.rb

```diff
@@ -25,19 +33,37 @@ def register_product(product_params)
   end
   
   def disp
-    puts "商品を選択してください"
+    puts "\n"
+    puts "商品を選んでください。"
+    puts "#{HR_STR}"
     @products.each do |product|
       puts "#{product.id}.#{product.name}" "(¥#{product.price})"
     end
+    puts "#{HR_STR}"
   end
 
   def ask_quantity(user)
     puts "#{user.selected_product.name}ですね、何個買いますか？"
+    puts "\n"
+  end
+
+  def calculate_charges(user)
+    total_price = user.quantity_of_product * user.selected_product.price #合計金額
```

Comment:
結果は同じですが，こちらの方が無難だと思います。

```suggestion
    total_price = user.selected_product.price * user.quantity_of_product #合計金額
```

細かいですが，一般にこういう話があるんですよね。

https://ja.wikipedia.org/wiki/%E3%81%8B%E3%81%91%E7%AE%97%E3%81%AE%E9%A0%86%E5%BA%8F%E5%95%8F%E9%A1%8C